pkg/updatestatus/controlplaneinformer.go

```diff
@@ -270,16 +459,35 @@ func versionsFromHistory(history []configv1.UpdateHistory) ControlPlaneUpdateVer
 	return versions
 }
 
+func parseQueueKey(queueKey string) (string, string, error) {
+	splits := strings.Split(queueKey, "/")
+	if len(splits) != 2 {
+		return "", "", fmt.Errorf("invalid queue key: %s", queueKey)
+	}
+	return splits[0], splits[1], nil
+}
+
 func configApiQueueKeys(object runtime.Object) []string {
 	if object == nil {
 		return nil
 	}
 
 	switch o := object.(type) {
 	case *configv1.ClusterVersion:
-		return []string{o.Name}
+		return []string{fmt.Sprintf("%s/%s", clusterVersionKindName, o.Name)}
+	case *configv1.ClusterOperator:
+		co := object.(*configv1.ClusterOperator)
+		for annotation := range co.Annotations {
+			if strings.HasPrefix(annotation, "exclude.release.openshift.io/") ||
+				strings.HasPrefix(annotation, "include.release.openshift.io/") {
```

Comment:
Hmm having this condition (essentially a filter for what COs we are interested in) is maybe a bit suprising in the function that is supposed to return queue keys for input objects. I think the right place for this is actually a filter - on L66 we would use a WithFilteredEventsInformersQueueKeyFunc instead:

https://github.com/openshift/cluster-version-operator/blob/e90705bb5457b2b9d449b4377036c3de6617ebb6/vendor/github.com/openshift/library-go/pkg/controller/factory/factory.go#L140-L154

This will allow us to decouple key-making logic (which should ideally be trivial) from what-cos-are-we-interested-in logic.

Reply:
Changed to WithInformersQueueKeysFunc to fit the signature: `Key` vs `Keys` in the name

https://github.com/openshift/cluster-version-operator/blob/e90705bb5457b2b9d449b4377036c3de6617ebb6/vendor/github.com/openshift/library-go/pkg/controller/factory/factory.go#L173

nit: 
I think we should use the singleton form functions `WithInformersQueueKeyFunc` and `WithFilteredEventsInformersQueueKeyFunc` as we generate only one key for an object in `configApiQueueKeys`. And then the library will convert it to the slice. Accordingly, `configApiQueueKeys` will be implemented as `configApiQueueKey` for this matter.

Reply:
The single-key methods use `ObjectQueueKeyFunc` which is deprecated though:

https://github.com/openshift/cluster-version-operator/blob/e90705bb5457b2b9d449b4377036c3de6617ebb6/vendor/github.com/openshift/library-go/pkg/controller/factory/factory.go#L73-L74

Reply:
Ha. I did not notice `ObjectQueueKeyFunc` is deprecated but not the functions using it as an arg such as `WithFilteredEventsInformersQueueKeyFunc`.
Thanks. I am all good.